src/main/java/com/kickstarter/dropwizard/metrics/influxdb/InfluxDbMeasurement.java

```diff
@@ -147,34 +152,85 @@ public Builder putTag(final String key, final String value) {
     /**
      * Adds the given key-value pair to the fields map.
      *
-     * @throws IllegalArgumentException if any value is not a String or primitive.
+     * @throws IllegalArgumentException if any value is not a String, primitive,
+     *         or Collection of Strings and primitives.
      */
     public <T> Builder putField(final String key, final T value) {
-      if (value == null) {
-        return this;
+      if (value instanceof Collection<?>) {
+        final Collection collection = (Collection) value;
+        final String collString = validatedPrimitiveCollection(key, collection);
+        fields.put(key, collString);
+      } else if (value != null) {
```

Comment:
as per the `@throws` in the doc comment above, should this throw if `value == null`?

Reply:
Invalid influxdb types like NaN/Inf/null are silently dropped; added some clarifying documentation on this behavior on the `putField/s` methods — not sure if adding warnings is helpful or would make things too noisy, considering how common I'd imagine `null` values to be.